libsel4/tools/cgenerator.py

```diff
@@ -0,0 +1,598 @@
+                CapType("seL4_X86_EPTPDPT", wordsize),
+                CapType("seL4_X86_EPTPD", wordsize),
+                CapType("seL4_X86_EPTPT", wordsize),
+                StructType("seL4_VCPUContext", wordsize * 7, wordsize),
+                StructType("seL4_UserContext", wordsize * 20, wordsize),
+            ],
+            "riscv32": [
+                Type("seL4_RISCV_VMAttributes", wordsize, wordsize),
+                CapType("seL4_RISCV_Page", wordsize),
+                CapType("seL4_RISCV_PageTable", wordsize),
+                CapType("seL4_RISCV_ASIDControl", wordsize),
+                CapType("seL4_RISCV_ASIDPool", wordsize),
+                StructType("seL4_UserContext", wordsize * 32, wordsize),
+            ],
+            "riscv64": [
+                Type("seL4_RISCV_VMAttributes", wordsize, wordsize),
+                CapType("seL4_RISCV_Page", wordsize),
+                CapType("seL4_RISCV_PageTable", wordsize),
+                CapType("seL4_RISCV_ASIDControl", wordsize),
+                CapType("seL4_RISCV_ASIDPool", wordsize),
+                StructType("seL4_UserContext", wordsize * 32, wordsize),
+            ]
+        }
+        return arch_types
+
+    @staticmethod
+    def generate_param_list(input_params, output_params):
+        # Generate parameters
+        params = []
+        for param in input_params:
+            if not param.type.pass_by_reference():
+                params.append(param.type.render_parameter_name(param.name))
+            else:
+                params.append(param.type.pointer().render_parameter_name(param.name))
+        for param in output_params:
+            if param.type.pass_by_reference():
+                params.append(param.type.pointer().render_parameter_name(param.name))
+
+        return ", ".join(params)
+
+    def _gen_file_header(self):
+        self.contents.append("#ifndef __LIBSEL4_SEL4_CLIENT_H")
+        self.contents.append("#define __LIBSEL4_SEL4_CLIENT_H")
+        self.contents.append("")
+
+        includes = ['autoconf.h', 'sel4/types.h']
+        self.contents.append('\n'.join(
+            ['#include <%s>' % include for include in includes]))
+
+        self.contents.append("""
+/*
+ * The following code generates a compile-time error if the system call
+ * stub generator has an incorrect understanding of how large a type is.
+ *
+ * If you receive a compile-time error here, you will need to adjust
+ * the type information in the stub generator.
+ */
+#define assert_size_correct(type, expected_bytes) \\
+        typedef unsigned long __type_##type##_size_incorrect[ \\
+                (sizeof(type) == expected_bytes) ? 1 : -1]
+""")
+        for x in self.data_types + self.arch_types[self.arch]:
+            self.contents.append("assert_size_correct(%s, %d);" % (x.name, x.native_size_bits / 8))
+        self.contents.append("")
+
+    def _gen_file_footer(self):
+        self.contents.append("#endif /* __LIBSEL4_SEL4_CLIENT_H */")
```

Comment:
Drop this, see https://github.com/seL4/seL4/pull/733/files#r777258403

Reply:
Done.